Rubberduck.Inspections/InspectionResult.cs

```diff
@@ -17,13 +20,22 @@ public InspectionResult(IInspection inspection, string description, Declaration
             : this(inspection, description, new QualifiedContext<ParserRuleContext>(target.QualifiedName.QualifiedModuleName, target.Context), target)
         { }
 
-        public InspectionResult(IInspection inspection, string description, QualifiedContext context, QualifiedMemberName? qualifiedMemberName)
+        public InspectionResult(IInspection inspection, string description, RubberduckParserState state, IdentifierReference reference)
+            : this(inspection, description, new QualifiedContext<ParserRuleContext>(reference.QualifiedModuleName, reference.Context), reference.Declaration, false)
+        {
+            QualifiedMemberName = GetQualifiedMemberName(state, reference);
+        }
+
+        public InspectionResult(IInspection inspection, string description, RubberduckParserState state, QualifiedContext context)
         {
             Inspection = inspection;
             Description = description.Capitalize();
             QualifiedName = context.ModuleName;
-            QualifiedMemberName = qualifiedMemberName;
+            QualifiedMemberName = GetQualifiedMemberName(state, context);
             Context = context.Context;
+
+            QualifiedSelection = new QualifiedSelection(QualifiedName, Context.GetSelection());
+            _navigationArgs = new Lazy<NavigateCodeEventArgs>(() => new NavigateCodeEventArgs(QualifiedSelection));
         }
 
         public InspectionResult(IInspection inspection, string description, QualifiedContext context, Declaration target, bool navigateToTarget = true)
```

Comment:
It's set to false any time we a) don't have a declaration (parse-tree inspection, for one), or b) any time we use a reference instead of a declaration.